Source/WebCore/page/scrolling/ScrollingTree.cpp

```diff
@@ -635,10 +635,13 @@ void ScrollingTree::setUserScrollInProgressForNode(ScrollingNodeID nodeID, bool
 {
     ASSERT(nodeID);
     Locker locker { m_treeStateLock };
-    if (isScrolling)
+    if (isScrolling) {
         m_treeState.nodesWithActiveUserScrolls.add(nodeID);
-    else
+        scrollingTreeNodeWillStartScroll(nodeID);
```

Comment:
On iOS, `ScrollingTreeScrollingNodeDelegateIOS::scrollWillStart()` calls `ScrollingTree().scrollingTreeNodeWillStartScroll()` so will there be two calls to `scrollingTreeNodeWillStartScroll()` on iOS now?

Reply:
hmm ya probably. I'll test to confirm. If so, should I just move this new logic into ScrollingTreeMac?

Reply:
![willstartscroll](https://user-images.githubusercontent.com/16073505/224514372-f4edf8ed-409f-4aba-8ebc-de7f8281d9bd.png)

Actually, I don't think it can ever get called twice on iOS, since `setUserScrollInProgressForNode` is not called on iOS afaik.

I also tested on device on apple.com and only ever saw the method get called once.